### DISCUSSION TITLE
AD5M w Klipper Mod - START_PRINT struggles — temp issues and 'slow crash' into bed. values come from ...?    :-)

### DISCUSSION THREAD

**Discussion by DougJoseph** [xblax/flashforge_ad5m_klipper_mod#57] [Q&A] [answered]
OK, so I installed the AD5M Klipper Mod. I am currently using the one with Klipper Screen.
 
I'm using OrcaSlicer, based on the Flashforge print profile for the DIY enclosure. 

I set the filament to PLA with a nozzle temp of 220 and a bed temp of 55 first layer and then 50.

Here is the GCode file that got outputted. The only edits were me adding some comments to more easily identify what each GCode does.

[3DBenchy_PLA_34m4s.gcode.zip](https://github.com/xblax/flashforge_adm5_klipper_mod/files/14579514/3DBenchy_PLA_34m4s.gcode.zip)

For START GCode I have:

`START_PRINT BED_TEMP={first_layer_bed_temperature[0]} EXTRUDER_TEMP={first_layer_temperature[0]}
`

For END GCode I have:

`END_PRINT
`

@Tiwatz says those settings are working there.

I noticed that although it wants to start by heating the nozzle to 255, later for actual printing the GCode calls for 220 instead. 

**Where does the  "first_layer_temperature[0]" value come from? How would I edit it?**

I also noticed that the bed temp gets set by the variable, and I cannot see where it ever gets changed. 

**Where does the  "first_layer_bed_temperature[0]" value come from? How would I edit it?**

Also, I have no idea why the nozzle was getting pressed so hard against the bed. **I think it bent my bed a little!** Any help on that would be so appreciated.

**Thanks for being so awesome, and for all your amazing work and knowledge.**

<img width="1728" alt="Screenshot 2024-03-12 at 4 31 42 PM" src="https://github.com/xblax/flashforge_adm5_klipper_mod/assets/8921397/91e2b482-071c-483d-b1cf-6240bceaf167">

"BEFORE EDITING"

<img width="934" alt="Screenshot 2024-03-12 at 4 32 13 PM" src="https://github.com/xblax/flashforge_adm5_klipper_mod/assets/8921397/8db136f6-05df-4b32-a23a-a0196906ec4f">

"AFTER EDITING"

<img width="935" alt="Screenshot 2024-03-12 at 4 39 44 PM" src="https://github.com/xblax/flashforge_adm5_klipper_mod/assets/8921397/241836d6-41ee-410d-90fa-d4da4b192dcf">

"THE PLA SETTING"

<img width="1728" alt="Screenshot 2024-03-12 at 5 50 11 PM" src="https://github.com/xblax/flashforge_adm5_klipper_mod/assets/8921397/0334b94a-ddbf-4b71-acce-4afe2b8d6267">

"GCODE SCREEN SHOT"

<img width="1728" alt="Screenshot 2024-03-12 at 6 44 42 PM" src="https://github.com/xblax/flashforge_adm5_klipper_mod/assets/8921397/d935445a-9768-40e2-9dbf-9184d9f3b7e8">

**Comment by DougJoseph** [comment, marked as answer]
So, I took another look at the "BEFORE" values, and based on them, I am going to try this:

`START_PRINT BED_TEMP=[bed_temperature_initial_layer_single] EXTRUDER_TEMP=[nozzle_temperature_initial_layer]`

UPDATE! That did it! The temps now show correctly.

<img width="1728" alt="Screenshot 2024-03-12 at 6 51 24 PM" src="https://github.com/xblax/flashforge_adm5_klipper_mod/assets/8921397/c23bb2fd-6783-45cd-a83b-b09f9ccee7b4">

Now I turn my attention to the bed crashing issue. 

UPDATE! OK, so I tried again (with the modified Param names) and the bed crash thing did not happen this time. Maybe a fluke. I'm calling this one solved! 

Thank you, everyone!!

**Reply by xblax** [reply]
That G-Code looks like it should work. Not sure why you had an issue with bed crashing. Maybe it makes sense to just try some movements from Mainsail/Fluidd and make sure everything is right. 

Make sure you have no bed mesh profile loaded and home the printer. Move the head to X,Y (0,0). Gradually move the bed up until it touches the nozzle. For me that happens at Z = -1.480

**Comment by adeadlydose** [comment]
So... My Klipper Screen Keeps throwing an error. 
unknown command:"G17"
I'm new to most G-code, kind of learning as I go. how would I go about fixing this?

**Reply by DougJoseph** [reply]
So, my understanding is that the error message is harmless, and it simply means that the instance of Klipper that is running, is not built new enough to understand a certain way of dealing with something. The errors can be ignored, and the there is some way to change how the slicer is outputting the Gcode to prevent the use of that particular code.

**Reply by KaruroChori** [reply]
https://github.com/xblax/flashforge_adm5_klipper_mod/blob/master/docs/SLICING.md#known-issues
In theory klipper was patched to accept it at some point but it never made to the public release.
If there were no problems with the patch I proposed, I think we should have it in the next release to avoid these messages since they will continue to pop up.

**Reply by adeadlydose** [reply]
Thanks for the input guys, I appreciate the fast responses. I did let it print because it didn't seem to be harming anything, and while i'm working on the z offset to get that perfect first layer. Everything seems to have turned out well on my benchy. I did decrease the extrusion rate by 5% while it was going in 1% increments. now i'm just trying to become familiar with klipper and how to do the basics with it.

**Reply by consp** [reply]
> In theory klipper was patched to accept it at some point but it never made to the public release.

That's not entirely true, the very first klipperscreen version had it, but afterwards it was removed (it's in my personal klipper copy).

> If there were no problems with the patch I proposed, I think we should have it in the next release to avoid these messages since they will continue to pop up.

I experienced no problems, though now generate gcode without it.

**Comment by arty121979** [comment]
Thank You so much 👍 ! Just bought as gift, from box on stock and updated firmware can print only these files what's in, tried from they slicers with zero result. Now installed kliper. I have question - there is possible to realise same start of the print like in original firmware ? I liked that simplicity first clean nozzle, cool down a bit untill pressed on pei and then start. Especially this is gift for 8 years old boy, I think he will struggle

**Comment by SMF1208** [comment]
Edit: This was user error, thank you for your help with this post!
